testplan/runnable/interactive/http.py

```diff
@@ -429,6 +445,57 @@ def put(self, test_uid, suite_uid, testcase_uid, param_uid):
                 param_group[param_uid] = new_testcase
                 return param_group[param_uid].serialize()
 
+    @api.route("/report/export")
+    class ExportReport(flask_restplus.Resource):
+        """
+        Interactive export endpoint. There is an API for exporting root
+        report object.
+        """
+
+        def get(self):
+            available_exporters = []
+            with ihandler.report_mutex:
+                for exporter in ihandler.exporters:
+                    available_exporters.append(exporter.name)
+            return {
+                "available": available_exporters,
+                "history": export_history,
+            }
+
+        @api.expect(post_export_model)
+        def post(self):
+            save_exports = request.json
+            with ihandler.report_mutex:
+                for exporter in ihandler.exporters:
+                    if exporter.name in save_exports.get("exporters", []):
+                        export_result = {
+                            "time": time.time(),
+                            "name": exporter.name,
+                            "uid": str(uuid.uuid4()),
+                        }
+                        try:
+                            exporter.export(ihandler.report)
+                            export_result["success"] = True
+                            export_result["message"] = exporter.export_path
```

Comment:
I think the intro of exporter.export_path and its usage in code is quite unnatural - can we change export() to return the exported report path? I think it will make the code look neater